public/app/features/alerting/unified/mocks/server/db.ts

```diff
@@ -0,0 +1,79 @@
+import { Factory } from 'fishery';
+
+import { DataSourceInstanceSettings, PluginType } from '@grafana/data';
+import { config, setDataSourceSrv } from '@grafana/runtime';
+import {
+  PromAlertingRuleDTO,
+  PromAlertingRuleState,
+  PromRuleGroupDTO,
+  PromRuleType,
+} from 'app/types/unified-alerting-dto';
+
+import { MockDataSourceSrv } from '../../mocks';
+import { DataSourceType } from '../../utils/datasource';
+
+const ruleFactory = Factory.define<PromAlertingRuleDTO>(({ sequence }) => ({
+  name: `test-rule-${sequence}`,
+  query: 'test-query',
+  state: PromAlertingRuleState.Inactive,
+  type: PromRuleType.Alerting,
+  health: 'ok',
+  labels: { team: 'infra' },
+}));
+
+const groupFactory = Factory.define<PromRuleGroupDTO>(({ sequence }) => {
+  const group = {
+    name: `test-group-${sequence}`,
+    file: `test-namespace`,
+    interval: 10,
+    rules: ruleFactory.buildList(10),
+  };
+
+  ruleFactory.rewindSequence();
+
+  return group;
+});
+
+const dataSourceFactory = Factory.define<DataSourceInstanceSettings>(({ sequence, params, afterBuild }) => {
+  afterBuild((dataSource) => {
+    config.datasources[dataSource.name] = dataSource;
+    setDataSourceSrv(new MockDataSourceSrv(config.datasources));
```

Comment:
I'm actually trying to deprecate the `MockDataSourceSrv` elsewhere, so it'd be good to use `setupDataSources` instead of this (we can discuss tomorrow anyway)